### ISSUE TITLE
PATH segment full style not displaying exactly what is saying

### ISSUE THREAD

**Issue by wyldspyder** [JanDeDobbeleer/oh-my-posh#329]
Hi,

I'm trying to display my path correctly into the path segment using the "full" style segment. the description say that its using the $PWD var in pwsh but seems not realy.

I've created some PSDrive ex:

set-localtion -name PSdir psprovider "FileSystem" root = "~\Documents\PowerShell\"

so when I am in the repertory, the $PWD say :

Path
----
PSdir:\

but the path segment continue to show the full path, ie:

C:\Users\<my user>\Documents\PowerShell

So it is normal or I've done something incorrectly ?? I know that we can use the "Mapped location" but it is kinda defeat the objective as i will need each time a new shortcut is createad to add it manually in my theme config

this is the config I user for the segment :

```
"type": "path",
"style": "powerline",
"powerline_symbol": "\uE0B0",
"foreground": "#000",
"background": "#42C6FF",
"properties": {
   "style": "full",
   "mapped_locations": [
      ["C:\\temp", "\ue799"]
   ]
```

### Environment

- Oh my Posh version: 3.68.4
- Theme: my own personal
- Operating System: windows 10
- Shell: windows terminal and VS powershell inegrated console

**Comment by JanDeDobbeleer** [comment]
@wyldspyder how did you import oh-my-posh into your `$PROFILE`? Are you using the module or a manual installation?
Also, I'm trying to execute the command in your example, but that doesn't seem to work (also doesn't map with the documentation):

```powershell
$ set-location -name PSdir psprovider "FileSystem" root = "~\Documents"

Set-Location : A parameter cannot be found that matches parameter name 'name'.
At line:1 char:14
+ set-location -name PSdir psprovider "FileSystem" root = "~\Documents"
+              ~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Set-Location], ParameterBindingException
    + FullyQualifiedErrorId : NamedParameterNotFound,Microsoft.PowerShell.Commands.SetLocationCommand
```

I think you mean `New-PSDrive`, so I'm going to debug using that one for now.

**Comment by JanDeDobbeleer** [comment]
Tested with PowerShell 7.2 using `New-PSDrive -Name PSdir -PSProvider "FileSystem" -Root "~\Documents"`. I can confirm this is by design. Under the hood we're using `$PWD.ProviderPath`. I can check if we want to distinct between visualising and actual location, but there were issues when running in the registry for example. I'll do a quick check.

**Comment by JanDeDobbeleer** [comment]
TL;DR, we can't use the current PWD flag as that needs to contain the ACTUAL path on the filesystem rather than a PS only PSDrive, else a lot of context can't be retrieved. But, I introduced a new flag specifically for this use case which is only used within the Powershell init script and module, that way we can separate the visual aspect from the functional one meaning any PSDrive will work from that version onwards.

**Comment by wyldspyder** [comment]
Hi JanDeDobbeleer, thanks fo rthe follow up, it is really appreciated !!

My importation of module automated, I will not show you everything but here the basic concept:
```

$moduleAvailable = get-installedModule

$moduleToImport = (
    @{
        name = "oh-my-posh" 
        allowPrerelease = $true
    })

foreach($module in $moduleToImport) {
    if ($moduleAvailable.contains($module.name)) {
        install-module @($module.name)
    }
}
```
Also, for importing all my 'drive', I use a similar concept :

```
$driveList = @{
    "ordiA"  = @{};
    "ordiB"  = @{}
}


# drive ordi A
$driveList.ordiA = (
    @{name = "Home";     psprovider = "FileSystem"; root = "~"},
    @{name = "GitHub";   psprovider = "FileSystem"; root = "~\Documents\GitHub\"},
    @{name = "Bitbk";    psprovider = "FileSystem"; root = "~\Documents\Bitbk\"}
)

# drive maison
$driveList.ordiB = (
    @{name = "Home";     psprovider = "FileSystem"; root = "~"},
    @{name = "PSdir";    psprovider = "FileSystem"; root = "~\Documents\PowerShell\"}
)

foreach($item in $driveList[$env:USERDOMAIN]){

    if ((test-path $item.root) -and (((get-psdrive).name) -notcontains $item.name)) {
        # créé le drive
        New-PSDrive @item | out-null

        #créé le raccouci "<name>:"
        invoke-expression ("function " + $item.name.tostring() + ': {set-location ' + $item.name + ':}')
    }
}

```

I d'ont think it is causing problem but I,ve just remarked that my PSh is at v7.1.0, I'll try with 7.2 just to be sure

**Comment by JanDeDobbeleer** [comment]
@wyldspyder the latest version should behave the way you want it. I extended the functionality to distinguish between visualization of the path (what do I show in the prompt) and functionality (where am I on the filesystem). Nothing wrong with your profile, any valid PSDrive should now also show as such in the latest version of oh-my-posh.
